service/src/main/java/bio/terra/workspace/service/admin/flights/cloudcontexts/gcp/SyncGcpIamRolesFlight.java

```diff
@@ -0,0 +1,45 @@
+package bio.terra.workspace.service.admin.flights.cloudcontexts.gcp;
+
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.IS_WET_RUN;
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.WORKSPACE_ID_TO_GCP_PROJECT_ID_MAP;
+import static java.util.Objects.requireNonNull;
+
+import bio.terra.stairway.Flight;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.RetryRule;
+import bio.terra.workspace.common.utils.FlightBeanBag;
+import bio.terra.workspace.common.utils.FlightUtils;
+import bio.terra.workspace.common.utils.RetryRules;
+import bio.terra.workspace.service.crl.CrlService;
+import com.fasterxml.jackson.core.type.TypeReference;
+import java.util.Map;
+import java.util.UUID;
+
+/** Flight to sync IAM roles of existing GCP projects. */
+public class SyncGcpIamRolesFlight extends Flight {
+
+  /** @InheritDoc */
+  public SyncGcpIamRolesFlight(FlightMap inputParameters, Object applicationContext) {
+    super(inputParameters, applicationContext);
+    FlightBeanBag appContext = FlightBeanBag.getFromObject(applicationContext);
+    CrlService crl = appContext.getCrlService();
+    Map<UUID, String> projectIds =
+        requireNonNull(
+            inputParameters.get(WORKSPACE_ID_TO_GCP_PROJECT_ID_MAP, new TypeReference<>() {}));
+
+    RetryRule cloudRetryRule = RetryRules.cloud();
+    addStep(new SetupWorkingMapForUpdatedWorkspacesStep());
+    for (Map.Entry<UUID, String> projectId : projectIds.entrySet()) {
+      // Wrap IAM with WSM service account.
+      addStep(
+          new RetrieveGcpIamCustomRoleStep(crl.getIamCow(), projectId.getValue()), cloudRetryRule);
+      if (FlightUtils.getRequired(inputParameters, IS_WET_RUN, Boolean.class)) {
+        // Update permissions for real only when it's not dry run.
+        addStep(
```

Comment:
The purpose of dry run is to see what would change. The only place we log this is GcpIamCustomRolePatchStep, right? If so, pass IS_WET_RUN to GcpIamCustomRolePatchStep. In GcpIamCustomRolePatchStep, log regardless of IS_WET_RUN.

Reply:
Could you add a section to DEVELOPMENT.md?

Only SAM admins can run this action.

- Dev dry run, check GCP logs to confirm new permissions/GCP projects as expected
- Dev wet run
- Autopush dry run
- Autopush wet run

Etc